src/PokeList.js

```diff
@@ -1,13 +1,56 @@
 import React, { Component } from 'react'
 import PokeItem from "./PokeItem.js";
+import request from 'superagent';
+import SearchBar from './SearchBar.js';
+import { Link } from 'react-router-dom'
+
 
 export default class PokeList extends Component {
+    state = {
+        searchQuery: this.props.match.params.name,
+        pokedex: [],
+    }
+
+    async componentDidMount() {
+            const data = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex/`)
+
+            this.setState({ pokedex: data.body.results })
+    }
+
+    handleSearch = async (e) => {
+        e.preventDefault();
+
+        const data = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex/?pokemon=${this.state.searchQuery}`)
+
+        this.setState({
+            pokedex: data.body.results,
+        })
+
+        this.props.history.push(this.state.searchQuery)
+    }
+
+handleChange = (e) => this.setState({ searchQuery: e.target.value })
+
     render() {
         return (
             <div>
-                <ul id="displayContainer">
-                    {this.props.data.map(pokemon => <PokeItem data={pokemon} />)}
+                <header>
+                    <SearchBar
+                    searchQuery={this.state.searchQuery}
+                    handleSearch={this.handleSearch}
+                    handleChange={this.handleChange}
+                    />
+                </header>
+                { this.state.pokedex.length &&
+                    <ul>
+                    {
+                        this.state.pokedex.map(character => <Link to={`pokemon/${character.pokemon}`}>
+                            <PokeItem data={character} />
```

Comment:
adding a key property here will fix your console error something like the id from the api